modules/miekg/miekg.go

```diff
@@ -1147,6 +1212,14 @@ func (s *Lookup) iterativeLookup(dnsType uint16, dnsClass uint16, name string, n
 }
 
 func (s *Lookup) DoMiekgLookup(name string) (interface{}, []interface{}, zdns.Status, error) {
+	if s.DNSType == dns.TypePTR {
+		var err error
+		name, err = dns.ReverseAddr(name)
+		if err != nil {
+			return nil, nil, zdns.STATUS_ILLEGAL_INPUT, err
+		}
+		name = name[:len(name)-1]
```

Comment:
Are you sure you want fall-through here?

Reply:
yeah, I tested it and it gets thrown back to the user correctly. I think it just feels weird because nowhere else do we do pre-processing like this.